backend/handlers/handlers.go

```diff
@@ -537,12 +537,16 @@ func (handlers *Handlers) postRenameAccountHandler(r *http.Request) (interface{}
 	type response struct {
 		Success      bool   `json:"success"`
 		ErrorMessage string `json:"errorMessage,omitempty"`
+		ErrorCode    string `json:"errorCode,omitempty"`
 	}
 
 	if err := json.NewDecoder(r.Body).Decode(&jsonBody); err != nil {
 		return response{Success: false, ErrorMessage: err.Error()}, nil
 	}
 	if err := handlers.backend.RenameAccount(jsonBody.AccountCode, jsonBody.Name); err != nil {
+		if errCode, ok := errp.Cause(err).(backend.ErrorCode); ok {
+			return response{Success: false, ErrorCode: errCode.Error()}, nil
```

Comment:
A bit confusing. I would expect an error code to already be some kind of machine-readable code. Wouldn't `ErrorCode: string(errCode)` be more logical here? ErrorCode satisfies the `error` type by implementing the `Error` method but it doesn't really mean you have to call it in all cases.

Reply:
done. fyi `Error()` returned the error code before, but `string(errCode)` is more robust.